sympy/tensor/array/tests/test_arrayop.py

```diff
@@ -84,3 +87,148 @@ def test_issue_emerged_while_discussing_10972():
+    assert permutedims(z, [2, 3, 1, 0]).shape == (6, 7, 5, 4)
+    assert permutedims(z, Permutation([2, 3, 1, 0])).shape == (6, 7, 5, 4)
+
+    po = Array(sa, [2, 2, 3, 3, 2, 2])
+
+    raises(ValueError, lambda: permutedims(po, (1, 1)))
+    raises(ValueError, lambda: po.transpose())
+    raises(ValueError, lambda: po.adjoint())
+
+    assert permutedims(po, reversed(range(po.rank()))) == Array(
+        [[[[[[sa[0], sa[72]], [sa[36], sa[108]]], [[sa[12], sa[84]], [sa[48], sa[120]]], [[sa[24],
+                                                                                           sa[96]], [sa[60], sa[132]]]],
+           [[[sa[4], sa[76]], [sa[40], sa[112]]], [[sa[16],
+                                                    sa[88]], [sa[52], sa[124]]],
+            [[sa[28], sa[100]], [sa[64], sa[136]]]],
+           [[[sa[8],
+              sa[80]], [sa[44], sa[116]]], [[sa[20], sa[92]], [sa[56], sa[128]]], [[sa[32],
+                                                                                    sa[104]], [sa[68], sa[140]]]]],
+          [[[[sa[2], sa[74]], [sa[38], sa[110]]], [[sa[14],
+                                                    sa[86]], [sa[50], sa[122]]], [[sa[26], sa[98]], [sa[62], sa[134]]]],
+           [[[sa[6],
+              sa[78]], [sa[42], sa[114]]], [[sa[18], sa[90]], [sa[54], sa[126]]], [[sa[30],
+                                                                                    sa[102]], [sa[66], sa[138]]]],
+           [[[sa[10], sa[82]], [sa[46], sa[118]]], [[sa[22],
+                                                     sa[94]], [sa[58], sa[130]]],
+            [[sa[34], sa[106]], [sa[70], sa[142]]]]]],
+         [[[[[sa[1],
+              sa[73]], [sa[37], sa[109]]], [[sa[13], sa[85]], [sa[49], sa[121]]], [[sa[25],
+                                                                                    sa[97]], [sa[61], sa[133]]]],
+           [[[sa[5], sa[77]], [sa[41], sa[113]]], [[sa[17],
+                                                    sa[89]], [sa[53], sa[125]]],
+            [[sa[29], sa[101]], [sa[65], sa[137]]]],
+           [[[sa[9],
+              sa[81]], [sa[45], sa[117]]], [[sa[21], sa[93]], [sa[57], sa[129]]], [[sa[33],
+                                                                                    sa[105]], [sa[69], sa[141]]]]],
+          [[[[sa[3], sa[75]], [sa[39], sa[111]]], [[sa[15],
+                                                    sa[87]], [sa[51], sa[123]]], [[sa[27], sa[99]], [sa[63], sa[135]]]],
+           [[[sa[7],
+              sa[79]], [sa[43], sa[115]]], [[sa[19], sa[91]], [sa[55], sa[127]]], [[sa[31],
+                                                                                    sa[103]], [sa[67], sa[139]]]],
+           [[[sa[11], sa[83]], [sa[47], sa[119]]], [[sa[23],
+                                                     sa[95]], [sa[59], sa[131]]],
+            [[sa[35], sa[107]], [sa[71], sa[143]]]]]]])
```

Comment:
Could you please tell me how do you generate such long expressions? I hope you don't write them by hand=)